lib/collection/src/shards/transfer/snapshot.rs

```diff
@@ -158,7 +159,7 @@ pub(super) async fn transfer_snapshot(
     remote_shard: RemoteShard,
     channel_service: ChannelService,
     consensus: &dyn ShardTransferConsensus,
-    snapshots_path: &Path,
+    snapshot_manager: SnapshotManager,
```

Comment:
Why `snapshot_manager` is required here as an argument, while we have access to `shard_holder` and `shard_id`?

Reply:
AFAICT, `shard_holder` doesn't have a path to access any pre-existing `SnapshotManager` instance, or a config struct. It cannot instantiate a `SnapshotManager` on its own.

Reply:
`shard_holder` have shards, and shards have config